grub-core/loader/emu/linux.c

```diff
@@ -33,6 +32,201 @@ static char *kernel_path;
 static char *initrd_path;
 static char *boot_cmdline;
 
+static grub_err_t
+grub_switch_root (void)
+{
+  char* options_cmd = NULL;
```

Comment:
Declarations use C-style, not C++-style.  (i.e., `char *options_cmd = NULL;`, not `char* options_cmd`.